apps/web/src/components/quick-start/in-app-onboarding/TriggerNode.tsx

```diff
@@ -76,39 +76,35 @@ function TriggerButton({ setOpened }: { setOpened: (value: boolean) => void }) {
   });
 
   const onboardingNotificationTemplate = templates.find((template) => template.name.includes(notificationTemplateName));
-
-  useEffect(() => {
-    async function createOnBoardingTemplate() {
-      const payloadToCreate = {
-        notificationGroupId: groups[0]._id,
-        isBlueprint: false,
-        name: notificationTemplateName,
-        active: true,
-        draft: false,
-        steps: [
-          {
-            template: {
-              type: StepTypeEnum.IN_APP,
-              content: 'Test notification {{number}}',
-              actor: {
-                type: ActorTypeEnum.SYSTEM_ICON,
-                data: SystemAvatarIconEnum.SUCCESS,
-              },
+  const hasToCreateOnboardingTemplate =
+    !templatesLoading && !notificationGroupLoading && !createTemplateLoading && !onboardingNotificationTemplate;
+
+  useEffectOnce(async () => {
+    const payloadToCreate = {
+      notificationGroupId: groups[0]._id,
+      isBlueprint: false,
+      name: notificationTemplateName,
+      active: true,
+      draft: false,
+      steps: [
+        {
+          template: {
+            type: StepTypeEnum.IN_APP,
+            content: 'Test notification {{number}}',
+            actor: {
+              type: ActorTypeEnum.SYSTEM_ICON,
+              data: SystemAvatarIconEnum.SUCCESS,
             },
           },
-        ],
-      };
-
-      await createNotificationTemplate({
-        template: payloadToCreate as unknown as ICreateNotificationTemplateDto,
-        params: { __source: TemplateCreationSourceEnum.ONBOARDING_IN_APP },
-      });
-    }
+        },
+      ],
+    };
 
-    if (!templatesLoading && !notificationGroupLoading && !createTemplateLoading && !onboardingNotificationTemplate) {
-      createOnBoardingTemplate();
-    }
-  }, [templates, onboardingNotificationTemplate]);
+    await createNotificationTemplate({
+      template: payloadToCreate as unknown as ICreateNotificationTemplateDto,
+      params: { __source: TemplateCreationSourceEnum.ONBOARDING_IN_APP },
+    });
+  }, hasToCreateOnboardingTemplate);
```

Comment:
this was the `useEffect` before, but the intention was to run the code only once when the condition is true, that's why I've changed it to `useEffectOnce`

Reply:
Love it :)
small note, i know the code was like this before but maybe we could extract the payloadToCreate outside so it will be easier to read, and I think we could remove the `await` for the creation because useEffectOnce is not async by design.

Reply:
There is no sense to extract the payload, otherwise, it will be created on every render call. But agree we don't need async/await here.